packages/entity-server/src/routes/hierarchy/types.ts

```diff
@@ -15,8 +15,27 @@ export interface HierarchyRequestParams {
 
 export type HierarchyRequestBody = Record<string, unknown>;
 
-export interface HierarchyRequestQuery {
+type FlattenAndPrefix<T, K extends keyof T & string> = {
+  [V in K]: { [field in keyof T[V] & string as `${V}.${field}`]: T[V][field] };
+}[K];
+
+type ObjectLikeKeys<T> = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [K in keyof T]: T[K] extends Record<string, any> ? K : never;
+}[keyof T];
+
+type SimpleFieldKeys<T> = {
+  [K in keyof T]: T[K] extends string | number | symbol ? K : never;
+}[keyof T];
+
+type EntityFilterQuery = Partial<
+  Omit<EntityFields, ObjectLikeKeys<EntityFields>> &
+    FlattenAndPrefix<EntityFields, ObjectLikeKeys<EntityFields>>
+>;
+
+export interface HierarchyRequestQuery extends EntityFilterQuery {
   fields?: string;
+  field?: string;
```

Comment:
Hope these params aren't too confusing... `field` is for getting the flat response, whereas `fields` will return an object with listed fields. I thought about just using `fields` and if just 1 is supplied we flatten it? My issue there would be it could be unexpected for clients to get a different data shape based on the number of requested fields...

Also, while the param in the query is `field` I'm calling it `flat` on the context object and all throughout the code. I did this since I fould field and fields to be rather confusing but I am happy to get pushback on it

Reply:
You know what you are right actually, and given that we are using typescript and they have different type signatures I don't think it'll be too confusing. I'll change flat->field internally as well 👍